test/test-generate-request-details.js

```diff
@@ -243,4 +244,26 @@ suite('Test Generate Request Details', function() {
     assert.equal(details.headers.Topic, extraOptions.headers.Topic);
     assert.equal(details.headers.Urgency, extraOptions.headers.Urgency);
   });
+
+  test('Audience contains port', function() {
+    const subscription = {
+      endpoint: 'http://example.com:4242/life-universe-and-everything'
+    };
+
+    const extraOptions = {
+      vapidDetails: {
+        subject: 'mailto:example@example.com',
+        publicKey: vapidKeys.publicKey,
+        privateKey: vapidKeys.privateKey
+      }
+    };
+
+    const message = undefined;
+    const requestDetails = webPush.generateRequestDetails(subscription, message, extraOptions);
+    const authHeader = requestDetails.headers.Authorization;
+    const audience = jws.decode(authHeader.slice(8)).payload.aud;
```

Comment:
I am decoding the JWT without verifying the signature, since it's coming from a trusted source, to get the decoded contents and then verify the `aud` value with the expected one.